zebra-network/src/peer_set/initialize/tests/vectors.rs

```diff
@@ -133,3 +144,78 @@ async fn local_listener_port_with(listen_addr: SocketAddr, network: Network) {
         "IP addresses are correctly propagated"
     );
 }
+
+/// Test if the initial seed peer connections is rate-limited.
+#[tokio::test]
+async fn add_initial_peers_is_rate_limited() {
+    zebra_test::init();
+
+    // We don't need to actually connect to the peers; we only need to check
+    // if the connection attempts is rate-limited. Therefore, just return an error.
+    let outbound_connector =
+        service_fn(|_| async { Err("test outbound connector always returns errors".into()) });
+
+    const PEER_COUNT: usize = 10;
+
+    let before = Instant::now();
+
+    let (crawl_task_handle, peerset_rx) = spawn_add_initial_peers(PEER_COUNT, outbound_connector);
+    let connections = peerset_rx.take(PEER_COUNT).collect::<Vec<_>>().await;
+
+    let elapsed = Instant::now() - before;
+
+    assert_eq!(connections.len(), PEER_COUNT);
+    // Make sure the rate limiting worked by checking if it took long enough
+    assert!(
+        elapsed > constants::MIN_PEER_CONNECTION_INTERVAL.saturating_mul((PEER_COUNT - 1) as u32),
+        "elapsed only {:?}",
+        elapsed
+    );
+
+    let _ = crawl_task_handle.await;
```

Comment:
Test Coverage: Let's also check for errors or panics in the initial peers task.

(If we ignore the task result, the tests will pass even if there are panics in the task, or errors returned by the task.)

I also fixed the name of the `JoinHandle` to match the task we're spawning.

```suggestion
    let (initial_peers_task_handle, peerset_rx) = spawn_add_initial_peers(PEER_COUNT, outbound_connector);
    let connections = peerset_rx.take(PEER_COUNT).collect::<Vec<_>>().await;

    let elapsed = Instant::now() - before;

    assert_eq!(connections.len(), PEER_COUNT);
    // Make sure the rate limiting worked by checking if it took long enough
    assert!(
        elapsed > constants::MIN_PEER_CONNECTION_INTERVAL.saturating_mul((PEER_COUNT - 1) as u32),
        "elapsed only {:?}",
        elapsed
    );

    let initial_peers_result = initial_peers_task_handle.await;
    assert!(
        matches!(initial_peers_result, Ok(Ok(_)))
        "unexpected error or panic in add_initial_peers task: {:?}",
        initial_peers_result,
    );
```

Reply:
I pushed a commit with this fix to the PR so we could merge.